packages/components/src/components/SegmentedControl/styles/segmentedControl.module.scss

```diff
@@ -164,7 +164,8 @@ $radius-without-border: calc(var(--radius) - var(--line-width));
     }
 
     // Show the active indicator when the state is on
-    .item[data-state='on'] ~ & {
+    .item[data-state='on'] ~ &,
```

Comment:
Since the tooltip and the segmented control both set 'data-state' I had to rely on aria-checked for the segmented-control state in those cases.